etcdserver/apply.go

```diff
@@ -788,3 +803,17 @@ func isGteRange(rangeEnd []byte) bool {
 func noSideEffect(r *pb.InternalRaftRequest) bool {
 	return r.Range != nil || r.AuthUserGet != nil || r.AuthRoleGet != nil
 }
+
+func removeNeedlessRangeReqs(txn *pb.TxnRequest) {
+	f := func(ops []*pb.RequestOp) {
+		for i := 0; i < len(ops); i++ {
```

Comment:
thanks, it seems to be better. I'll follow this style and remove the nil checks in the apply phase.